modules/mongodb/mongodb.go

```diff
@@ -176,17 +191,27 @@ func initiateReplicaSet(req *testcontainers.GenericContainerRequest, cli mongoCl
 		req.LifecycleHooks, testcontainers.ContainerLifecycleHooks{
 			PostStarts: []testcontainers.ContainerHook{
 				func(ctx context.Context, c testcontainers.Container) error {
-					ip, err := c.ContainerIP(ctx)
+					// Wait for MongoDB to be ready
+					if err := waitForMongoReady(ctx, c, cli); err != nil {
+						return fmt.Errorf("wait for mongo: %w", err)
+					}
+
+					// Initiate replica set
+					host, err := c.Host(ctx)
+					if err != nil {
+						return fmt.Errorf("get host: %w", err)
+					}
+					mappedPort, err := c.MappedPort(ctx, "27017/tcp")
 					if err != nil {
-						return fmt.Errorf("container ip: %w", err)
+						return fmt.Errorf("get mapped port: %w", err)
 					}
 
 					cmd := cli.eval(
-						"rs.initiate({ _id: '%s', members: [ { _id: 0, host: '%s:27017' } ] })",
+						"rs.initiate({ _id: '%s', members: [ { _id: 0, host: '%s:%s' } ] })",
 						replSetName,
-						ip,
+						host,
+						mappedPort.Port(),
```

Comment:
question: why do we need the mapped port here? The command mongoCli's eval command executes **inside the container**, so the listening port must be 27017 instead of the exposed port in the host.

I tried locally using the old cli command, and removing the code in the tests that calculated the network interfaces, setting the `TESTCONTAINERS_HOST_OVERRIDE` env var, and all the tests pass again.

Reply:
@ttruongatl I still see this error when trying to push to your branch:

```
fatal: The upstream branch of your current branch does not match
the name of your current branch.  To push to the upstream branch
on the remote, use

    git push upstream HEAD:refs/pull/2984/head

To push to the branch of the same name on the remote, use

    git push upstream HEAD

To avoid automatically configuring an upstream branch when its name
won't match the local branch, see option 'simple' of branch.autoSetupMerge
in 'git help config'.
```

I'm attaching a PATCH file so you can add it to your PR.

[0001-chore-simplify-tests.patch](https://github.com/user-attachments/files/19433212/0001-chore-simplify-tests.patch)

Cheers!